hera_pspec/pspecdata.py

```diff
@@ -727,42 +765,88 @@ def scalar(self, stokes='I', taper='none', little_h=True, num_steps=2000):
                 numerical integral
                 Default: 10000
 
+        spw_select : tuple, Example: (200, 300)
+            tuple containing a start and stop channel used to index the `freq_array` of the datasets.
+            The default (None) is to use the entire band provided in the dataset.
+
+        beam : PSpecBeam object
+            Option to use a manually-fed PSpecBeam object instead of using self.primary_beam.
+
         Returns
         -------
         scalar: float
                 [\int dnu (\Omega_PP / \Omega_P^2) ( B_PP / B_P^2 ) / (X^2 Y)]^-1
                 in h^-3 Mpc^3 or Mpc^3.
         """
-        scalar = self.primary_beam.compute_pspec_scalar(self.freqs[0], self.freqs[-1], len(self.freqs), stokes=stokes,
-                                                        taper=taper, little_h=little_h, num_steps=num_steps)
+        # set spw_select and get freqs
+        if spw_select is None:
+            spw_select = (0, self.freqs.shape[1])
+        freqs = self.freqs[spw_select[0]:spw_select[1]]
+
+        # calculate scalar
+        if beam is None:
+            scalar = self.primary_beam.compute_pspec_scalar(freqs[0], freqs[-1], len(freqs), stokes=stokes,
+                                                            taper=taper, little_h=little_h, num_steps=num_steps)
+        else:
+            scalar = beam.compute_pspec_scalar(freqs[0], freqs[-1], len(freqs), stokes=stokes,
+                                                            taper=taper, little_h=little_h, num_steps=num_steps)
 
         return scalar
 
-    def pspec(self, bls, beam=None, input_data_weight='identity', norm='I', 
-              taper='none', little_h=True, verbose=True):
+    def pspec(self, bls1, bls2, dset1_ind, dset2_ind, input_data_weight='identity', norm='I', taper='none', 
```

Comment:
The `dset1_ind` keyword is explicit, but also a bit clunky. Can we get away with just `dset1` and `dset2`?

Reply:
Ah, no. How about `dsets=(dset1, dset2)`?

Reply:
yup